server/src/main/java/org/opensearch/action/search/PitService.java

```diff
@@ -129,4 +139,40 @@ public void onFailure(final Exception e) {
             }
         }, size);
     }
+
+    public void getAllPits(ActionListener<GetAllPitNodesResponse> getAllPitsListener) {
+        final List<DiscoveryNode> nodes = new LinkedList<>();
```

Comment:
Why `LinkedList`? Also, we can directly use streams and collect it into list by combining this statement and next for loop.